Utils/Dataflow/pyDKB/dataflow/stage/AbstractProcessorStage.py

```diff
@@ -309,8 +309,12 @@ def stream_input(self, fd):
             iterator = iter(fd.readline, "")
         else:
             iterator = custom_readline(fd, self.ARGS.eom)
-        for line in iterator:
+        try:
+            for i, line in enumerate(iterator):
                 yield self.parseMessage(line)
+        except KeyboardInterrupt:
+            sys.stderr.write("(WARN) Input stream processing interrupt"
+                             " on line %s\n" % i)
```

Comment:
`KeyboardInterrupt` (raised after pressing `Ctrl+C` in terminal, for example) is meant to stop the program (in a more or less gentle way), right?

Now let`s see what happens if the exception was caught here.

1. The `stream_input()` comes to its end, so `StopIteration` is raised.
2. This `StopIteration` stops loop in `run()` method, which then returns nothing (`None`, to be precise):
https://github.com/PanDAWMS/dkb/blob/48599f0b26af1a1de121c43d9ada5cc4a616497e/Utils/Dataflow/pyDKB/dataflow/stage/AbstractProcessorStage.py#L229-L240
3. Generally it means that the main program will also come to its end, like it does here:
https://github.com/PanDAWMS/dkb/blob/48599f0b26af1a1de121c43d9ada5cc4a616497e/Utils/Dataflow/054_docContent2TTL/docContent2TTL.py#L161-L171
  ...but nobody said that the one who uses pyDKB library can\`t do something else: e.g., restart processing in a `while True` loop. If one does, then pressed `Ctrl+C` will interrupt processing just to restart it again, and again -- till user gives up and kills the program in a less gently way. I met such a behaviour and it was a bit annoying ;)

Of course, the villain is the one who added `while True` loop aroun `run()` method call ;) Still I think we must let him know that the program was somehow interrupted, **not losing or masking** this event somewhere inside the library.

---

Actually, this is a question I wanted to discuss concerning [this](904674a284df43e629b055d20987630d144b25a6) commit of mine (see PR #99 ), which does much the same as 48599f0 here -- but in `run()` method. "Closer to the edge", so to speak :)

If we want to output warning/information message here (which is fair enough), the `try/except` block is on its proper place, in both cases (in PR #99 it ends up in `output_error()` method of `AbstractStage`, that outputs traceback OR says "Interrupted by user" in case of `KeyboardInterrupt`).
But what do we do after this, if we want to ensure the main program that uses `pyDKB` will also be interrupted?
We can re-raise `KeyboardInterrupt` (or any other `BaseException` (not `Exception`) caught), after it was handled some way in the library, or raise `SystemExit` (same to call `sys.exit()`), or...?

In the #99 I chose to use `sys.exit()`, so that the program ends, whatever is in the main code after `run()` method call (if no one catches `SystemExit`, of course):
https://github.com/PanDAWMS/dkb/blob/904674a284df43e629b055d20987630d144b25a6/Utils/Dataflow/pyDKB/dataflow/stage/AbstractProcessorStage.py#L246-L260

The idea was to screen final user from any traceback output in those cases when it is obvious that the program must die (namely, the `KeyboardInterrupt` case) -- but provide more or less uniform traceback output in other cases. And by doing this here, in the library -- save the main program developer that uses `pyDKB` from solving this problem again and again.

The only thing is that I am not sure it was done good enough, that nothing was forgotten of missed, and that the logic is not inverted somehow :)

---

Concerning this PR (sorry for the long lyrical digression):
1. `KeyboardInterrupt` is masked now: code outside `stream_input()` doesn`t know, if the input data are over or the program was interrupted by user. Both variants look the same now. Was it intented?
1. I see three possible ways to improve the situation:
    * re-raise exception
    *(then after #99 there will be two messages about "Interrupted by user.", but I will fix it after merging actual version to #99)*
    * call `sys.exit()` right here 
    *(then for now the program will simply exit here -- looks good enough for me -- and it will become a good reason to properly handle `SystemExit` in #99)*
    * leave `KeyboardInterrupt` unhandled and wait for #99
    *(which is just not fair, since you`ve noticed that it is unhandled, thought of how to fix it AND read to the end of this comment ;) )*

Reply:
@mgolosova 
Thank you for all comments. I'll fix everything tomorrow.
____________
Asking to your questions:
1. No. It wasn't intented.
2. Also, I prefer to call `sys.exit()` as it was in your second offer. :) Should I call it in added `KeyboardInterrupt`?

Reply:
Sorry, missed your reply.

@anastasiakaida wrote:
> Should I call it in added KeyboardInterrupt?

You mean, in the `except` part, where the `KeyboardInterrupt` was caught? Yes, output something to explain why the program has stopped and call `exit()`.
It seems to me that it will be Ok to exit with the default return value, as it is a regular situation -- so no need to return some error code.